sed/calibrator/delay.py

```diff
@@ -25,22 +28,102 @@ class DelayCalibrator:
+        self.fluctuations: Dict[str, Any] = self._config["delay"].get("fluctuations", {})
 
     def append_delay_axis(
+        self,
+        df: Union[pd.DataFrame, dask.dataframe.DataFrame],
+        *args,
+        **kwargs,
+    ) -> Tuple[Union[pd.DataFrame, dask.dataframe.DataFrame], dict]:
+        """TODO: docstring"""
+        if self.loader not in ["mpes", "flash"]:
+            raise NotImplementedError(
+                f"Delay calibration is implemented for 'mpes' and 'flash', not {self.loader}.",
+            )
+        method = getattr(self, f"append_delay_axis_{self.loader}")
+        return method(df, *args, **kwargs)
+
+    def append_delay_axis_flash(
+        self,
+        df: Union[pd.DataFrame, dask.dataframe.DataFrame],
+        time0: float = None,
+        flip_time_axis: bool = None,
+        delay_stage_column: str = None,
+        delay_column: str = None,
+        **kwargs,
+    ) -> Tuple[dask.dataframe.DataFrame, dict]:
+        """Calculate and append the delay axis to the events dataframe.
+
+        Args:
+            df (Union[pd.DataFrame, dask.dataframe.DataFrame]): The dataframe where
+                to apply the delay calibration to.
+            time0 (float, optional): Pump-Probe overlap value of the delay coordinate.
+                If omitted, it is searched for in the data files.
+            flip_time_axis (bool, optional): Invert the time axis.
+            delay_stage_column (str, optional): Source column for delay calibration.
+                Defaults to config["dataframe"]["delay_stage_column"].
+            delay_column (str, optional): Destination column for delay calibration.
+                Defaults to config["dataframe"]["delay_column"].
+
+        Returns:
+            Union[pd.DataFrame, dask.dataframe.DataFrame]: dataframe with added column
+            and delay calibration metdata dictionary.
+        """
+        if len(kwargs) > 0:
+            print(f"WARNING: arguments {kwargs.keys()} are not used in flash delay calibration.")
+        assert self.loader == "flash", "Invalid loader for this method."
+        # pylint: disable=duplicate-code
+        if delay_stage_column is None:
+            delay_stage_column = self.delay_stage_column
+        if delay_column is None:
+            delay_column = self.delay_column
+
+        if time0 is None:
+            time0 = self.calibration["time0"]
+        else:
+            self.calibration["time0"] = time0
+        if time0 is None:
+            raise ValueError("No time0 value specified.")
+
+        if flip_time_axis is None:
+            fta_str: str = str(self.calibration.get("flip_time_axis", False))
+            flip_time_axis = fta_str.lower() in ["true", "1", "yes"]
+        else:
+            self.calibration["flip_time_axis"] = flip_time_axis
+
+        def calibrate_time(x, time0, flip_time_axis) -> Any:
+            return (x[delay_stage_column] - time0) * (-1 if flip_time_axis else 1)
```

Comment:
can you use the shift_by_constant function we already have in dfops? It can also do a flip, no?